machineconfiguration/v1/types_machineosbuild.go

```diff
@@ -0,0 +1,198 @@
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+	// BuilderReference describes which ImageBuilder backend to use for this build
+	// +optional
+	BuilderReference *MachineOSBuilderReference `json:"builderReference"`
+	// relatedObjects is a list of references to ephemeral objects such as ConfigMaps or Secrets that are meant to be consumed while the build process runs.
+	// After a successful build or when this MachineOSBuild is deleted, these ephemeral objects should be deleted.
+	// However, in the event of a failed build, the objects will not be deleted to allow for inspection and debugging of the failed build process.
+	// +kubebuilder:validation:MaxItems=10
+	// +listType=map
+	// +listMapKey=name
+	// +listMapKey=resource
+	RelatedObjects []ObjectReference `json:"relatedObjects,omitempty"`
+	// buildStart describes when the build started.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="buildStart is immutable once set"
+	// +kubebuilder:validation:Required
+	BuildStart metav1.Time `json:"buildStart"`
+	// buildEnd describes when the build ended.
+	// When omitted the build has either not been started, or is in progress.
+	// It will be populated once the build completes, fails or is interrupted.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="buildEnd is immutable once set"
+	// +optional
+	BuildEnd *metav1.Time `json:"buildEnd,omitempty"`
+	// finalImagePushSpec describes the fully qualified pushspec produced by this build that the final image can be. Must end with a valid '@sha256:<digest>' suffix, where '<digest>' is 64 hexadecimal characters long
+	// +kubebuilder:validation:XValidation:rule=`((self.split('@').size() == 2 && self.split('@')[1].matches('^sha256:[a-f0-9]{64}$')))`,message="the OCI Image reference must end with a valid '@sha256:<digest>' suffix, where '<digest>' is 64 hexadecimal characters long"
+	// +optional
+	FinalImagePushspec string `json:"finalImagePushspec,omitempty"`
+}
+
+// MachineOSBuilderReference describes which ImageBuilder backend to use for this build
+type MachineOSBuilderReference struct {
+	// ImageBuilderType describes the image builder set in the MachineOSConfig, which in turn describes the builder that the cluster will attempt the build with.
+	// Currently only JobImageBuilder is supported, which will spin up a custom pod builder that uses buildah to build the specified image.
+	// +kubebuilder:validation:Required
+	ImageBuilderType MachineOSImageBuilderType `json:"imageBuilderType"`
+
+	// ImageBuilderRef is a reference to the object that is managing the image build
+	// For example, if the imageBuilderType is JobImageBuilder, this will be a reference to the Job object managing the build
+	// +optional
+	ImageBuilderRef *ObjectReference `json:"ImageBuilderRef,omitempty"`
```

Comment:
While I happen to be looking at this field, API conventions say you shouldn't use `Ref`, and therefore just ImageBuilder. Also, your json tag is not camelCase, it's PascalCase, that needs to be fixed